### PR TITLE
Match ``rdb.c``, ``mips.c``, ``flash.c`` and ``video.c``

### PR DESCRIPTION
**Pull request by Yanis42**
they're all the same except get32/put32 functions, the only one I didn't match is ``rdbPut32`` because I don't understand m2c's output, it's very confusing there's a switch in a switch in a switch, I'm completely lost on this one lol

### CODE REVIEW

File: src/flash.c [hunk 1]
```diff
@@ -7,24 +12,175 @@ _XL_OBJECTTYPE gClassFlash = {
...
-#pragma GLOBAL_ASM("asm/non_matchings/flash/flashPut64.s")
+    if (!ramGetBuffer(SYSTEM_RAM(pFLASH->pHost), &pTarget, nOffsetRAM, (u32*)&nSize)) {
+        return 0;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/flash/flashGet8.s")
+    switch (pFLASH->flashCommand & 0xFF000000) {
+        case 0xB4000000:
+            for (i = 0; i < nSize; i++) {
+                pFLASH->flashBuffer[i] = ((char*)(pTarget))[i];
```
Comment (cadmic) [inline src/flash.c:60]:
```suggestion
                pFLASH->flashBuffer[i] = ((char*)pTarget)[i];
```

File: src/flash.c [hunk 2]
```diff
@@ -7,24 +12,175 @@ _XL_OBJECTTYPE gClassFlash = {
...
+        case 0x4B000000:
+        case 0xD2000000:
+        default:
+            break;
+        case 0x78000000:
+            pFLASH->flashStatus = 0x11118008;
+            for (i = 0; i < ARRAY_COUNT(buffer); i++) {
+                buffer[i] = 0;
+            }
+            if ((pFLASH->flashCommand & 0xFF000000) + 0xC4000000 == 0) {
```
Comment (cadmic) [inline src/flash.c:102]: for these `x + y = 0` conditionals, can they be rewritten as `x = y'`? Similar to the movie night check

File: src/flash.c [hunk 3]
```diff
@@ -7,24 +12,175 @@ _XL_OBJECTTYPE gClassFlash = {
...
+                for (i = 0; i < 1024; i++) {
+                    if (!simulatorWriteFLASH(i << 7, (u8*)buffer, ARRAY_COUNT(buffer))) {
+                        return 0;
+                    }
+                }
+            } else {
+                if (((pFLASH->flashCommand & 0xFF000000) + 0xB5000000 == 0) &&
+                    !simulatorWriteFLASH((pFLASH->flashCommand << 7) & 0x7FFFFF80, (u8*)buffer, ARRAY_COUNT(buffer))) {
```
Comment (cadmic) [inline src/flash.c:110]: I prefer
```
} else if ((pFLASH->flashCommand & 0xFF000000) + 0xB5000000 == 0) {
    if (!simulatorWriteFlash(...)) {
        return 0;
}
```

File: src/mips.c [hunk 4]
```diff
@@ -8,34 +8,222 @@ _XL_OBJECTTYPE gClassMips = {
...
+            }
+            if (nData & 0x2000) {
+                pMips->nMode |= 0x200;
+            }
+            break;
+        case 12:
+            nData = *pData & 0xFFF;
 
-#pragma GLOBAL_ASM("asm/non_matchings/mips/mipsResetInterrupt.s")
+            if (*pData & 1) {
```
Comment (cadmic) [inline src/mips.c:126]:
```suggestion
            if (nData & 1) {
```
(I hope)

File: src/video.c [hunk 5]
```diff
@@ -8,44 +10,203 @@ _XL_OBJECTTYPE gClassVideo = {
...
+            break;
+        case 0x20:
+            pVideo->nSyncLeap = *pData & 0x0FFFFFFF;
+            break;
+        case 0x24:
+            if ((pVideo->nStartH = *pData & 0x03FF03FF) == 0) {
+                if (pVideo->bBlack != 1) {
+                    pVideo->bBlack = 1;
+                }
+            } else if (pVideo->bBlack) {
```
Comment (cadmic) [inline src/video.c:73]: I slightly prefer splitting this `else if` for better parallel structure. Also does `pVideo->bBlack != 0` work to mirror the `!= 1` check?

File: src/rdb.c [hunk 6]
```diff
@@ -15,20 +16,58 @@ void* jtbl_800EE1C0[] = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/rdb/rdbGet16.s")
+static s32 rdbGet16(Rdb* pRDB, u32 nAddress, s16* pData) { return 0; }
 
-#pragma GLOBAL_ASM("asm/non_matchings/rdb/rdbGet32.s")
+static s32 rdbGet32(Rdb* pRDB, u32 nAddress, s32* pData) {
+    switch (nAddress & 0xF) {
+        case 0:
+        case 8:
+        case 12:
```
Comment (cadmic) [inline src/rdb.c:37]: write as hex? 0x0, 0x8, 0xC

File: src/video.c [hunk 7]
```diff
@@ -8,44 +10,203 @@ _XL_OBJECTTYPE gClassVideo = {
...
+            break;
+        case 0x28:
+            pVideo->nStartV = *pData & 0x03FF03FF;
+            break;
+        case 0x2C:
+            pVideo->nBurst = *pData & 0x03FF03FF;
+            break;
+        case 0x30:
+            pVideo->nScaleX = *pData & 0xFFF;
+            if (!frameSetSize(SYSTEM_FRAME(pVideo->pHost), 0, pVideo->nSizeX, (s32)(pVideo->nScaleY * 0xF0) / 1024)) {
```
Comment (cadmic) [inline src/video.c:85]:
```suggestion
            if (!frameSetSize(SYSTEM_FRAME(pVideo->pHost), 0, pVideo->nSizeX, (s32)(pVideo->nScaleY * 240) / 1024)) {
```
ditto below (recognizable as 240p N64 screen size)

**Review by cadmic — COMMENTED** [review]
I made some progress on rdbPut32, do you feel like matching that or do you want to get this in first?

File: src/flash.c (hunk 2 above)
Comment (Yanis42) [reply src/flash.c:102]: I'm not sure to follow, are you saying like ``(pFLASH->flashCommand & 0xFF000000) == 0xC4000000)``?

**Comment by Yanis42** [comment]
yea would be cool to include the remaining function, I'll try to match it

File: src/flash.c (hunk 2 above)
Comment (cadmic) [reply src/flash.c:102]: I think `(pFLASH->flashCommand & 0xFF000000) == 0x3C000000` in this case

Comment (Yanis42) [reply src/flash.c:102]: it works but I don't understand why it's 0x3C000000 and not 0xC4000000

**Comment by Yanis42** [comment]
so, I pushed new commits to the branch but it doesn't update here but it's on the branch, I don't know what's going on

**Review by cadmic — APPROVED** [review]

**Pull request merged**